corehq/apps/users/models.py

```diff
@@ -2338,15 +2337,19 @@ def update_device_id_last_used(self, device_id, when=None, commcare_version=None
                 self.last_device = DeviceIdLastUsed.wrap(self.get_last_used_device().to_json())
                 meta = self.last_device.get_last_used_app_meta()
                 self.last_device.app_meta = [meta] if meta else []
-                return True
+                save_user = True
+            if fcm_token and fcm_token_timestamp:
```

Comment:
Good point! 
I see they are doing a check on date rather than the timestamp. We could probably do the same check as token change is rare.  There is one thing I am worried about is - lets say user uninstalls and reinstalls the app on the same day , this will change the token which will not be saved onto the database until the next day.

Reply:
Got it!  `UserReportingMetadataStaging` is being updated from multiple places.
I think yes it is good to add the mentioned check `fcm_token_timestamp > device.fcm_token_timestamp`

Reply:
Resolved in 8022174fe13b0237b4deafb892882a873507a102